### DISCUSSION TITLE
Process non markdown files

### DISCUSSION THREAD

**Discussion by redtide** [fralau/mkdocs-macros-plugin#170] [Ideas]
Hello! This has already been discussed in mkdocs/mkdocs#3251, but looking at the template files I'm using, it would be much more simply solved if it could be possible to process macros' jinja templates in non markdown files:
`docs/assets/img/badge.svg`:
```jinja
{%- include "status_progress.j2" %}
{%- import "status_badge.j2" as status_badge with context %}
{{- status_badge.set(name, progress) }}
```
`includes/status_progress.j2` has the svg data plus some jinja template variables to generate it with the needed label (`name` parameter, e.g. "ARIA"), and progress bar value (`progress`, e.g. a variable with a value of 57).

`includes/status_badge.j2` has some macros that calculate the values based on a YAML file data. This template file is already used for a table in another page, so I reuse the code for these badge images.

Result (despite a font size issue):
<img src="https://sfz.tools/assets/img/sfizz/badge_aria.svg">

**Comment by fralau** [comment]
I m not exactly sure if I understand your question. Isn't it something that jinja2 provides out of the box? Both Markdown and jinja2 are largely agnostic: you can import non Markdown files. It will be gracefully ignored (unless it conflicts with Markdown syntax); and it if is HTML, it will keep it as-is.

Hence, you could, in principle, import whatever file of whatever format you want.

**Comment by redtide** [comment]
I was asking if is it possible via MkDocs or your plugin to process jinja2 template in files other than markdown, in a SVG in this case.
It's not much time since I started to use MkDocs, so it might be that I miss something, just I wasn't able to do it (I come from jekyll).

For the rest I tried to explain what I was going to do with that code; basically using some "includes" to process data to generate the final SVG code.

**Comment by fralau** [comment]
II understand: if you want to import an SVG, with a little more than the standard Markdown facilities, a macro would be a good fit.

I asked ChatGPT how to do that... and here it is its answer:

```
def define_env(env):
    @env.macro
    def display_svg(filename):
        with open(filename, 'r') as file:
            svg_content = file.read()
        return svg_content
```

Which is a little naive, because you want HTML, I guess. But you have the gist.

**Comment by redtide** [comment]
I want to process jinja2 code inside a svg to produce the svg in the same way markdown with jinja2 code inside produce the final html.

**Reply by fralau** [reply]
It sounds good: I suppose you want your function to return HTML code. You want to inline the SVG into the page, or make the HTML tag read the file?

**Reply by redtide** [reply]
I just want the "templated" svg processed and converted to a working svg just like a markdown with template in it is converted to html. IDK what you mean with html tag read the file. The svg is read from the html already if its there.

**Reply by fralau** [reply]
I suppose you want a macro (function) that generates this type of HTML code?

```html
<object data="foo/bar/baz.svg" type="image/svg+xml">
 </object>
```

If that's not the case, you should provide a description of your input data, the result code you want in the page, and an example.

**Reply by redtide** [reply]
No, this is related to mkdocs/mkdocs#3251. I just want to process a svg file with jinja2 template code inside to produce a final, usable svg file, but macros plugin should not care about its content or format, just process the jinja2 template in it like it does with markdown files with jinja code inside. Other than that, as example, do a similar thing with an atom.xml feed:

```xml
<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom">
	<title>{{ site.name }} News</title>
	<link href="{{ '/atom.xml' | absolute_url }}" rel="self"/>
	<icon>{{ '/assets/ico/favicon.ico' | absolute_url }}</icon>
	<link href="{{ '/' | absolute_url }}"/>
	<updated>{{ site.time | date_to_xmlschema }}</updated>
	<id>{{ '/' | absolute_url }}</id>
{% for post in site.posts limit:20 %}
	<entry>
		<title>{{ post.title }}</title>
		<author><name>{{ post.author }}</name></author>
		<link href="{{ post.url | absolute_url }}"/>
		<published>{{ post.date | date_to_xmlschema }}</published>
		<id>{{ post.id | absolute_url }}</id>
		<content type="html">{{ post.content | xml_escape }}</content>
	</entry>
{% endfor %}
</feed>
```
this in Jekyll/Liquid produces a feed with all posts files published in the produced xml after the site build.
I want to do the same thing with MKDocs and jinja2 but I don't know if it's possible. Hope is clear now.

**Reply by fralau** [reply]
See my suggestion on https://github.com/mkdocs/mkdocs/discussions/3251#discussioncomment-7855568

**Comment by redtide** [comment]
The discussion in MKDocs was solved, so closing this, thank you again for the tips!
